pkg/envoy/admin/client.go

```diff
@@ -1,116 +1,166 @@
+					cert, _ := x509.ParseCertificate(rawCerts[0]) // ignore error because cert was parsed already
+					dpSpiffe := xds_tls.ServiceSpiffeID(dataplane.Meta.GetMesh(), dataplane.Spec.GetIdentifyingService())
+					for _, uri := range cert.URIs {
+						if uri.String() == dpSpiffe {
+							return nil
+						}
+					}
+					return errors.Errorf("could not find expected URI SAN %s", dpSpiffe)
+				},
+				// We disable builtin verification because
+				// 1) In first case, we don't have stable self-signed cert between instances of CP and we don't want to operate them.
+				// 2) it expects hostname or IP in cert instead of SPIFFE URI SAN, so we cannot use builtin verification
+				//
+				// Also keep in mind that on this very moment we are not sending sensitive data to the DP.
+				InsecureSkipVerify: true,
+				Certificates:       []tls.Certificate{a.clientCert},
+			},
+		},
+		Timeout: 5 * time.Second,
+	}
+	return c, err
 }
 
-func (a *envoyAdminClient) getOrCreateSigningKey(mesh string) (string, error) {
-	key, err := issuer.GetSigningKey(a.rm, issuer.DataplaneTokenPrefix, mesh)
+func (a *envoyAdminClient) caCertPoolOfMeshMTLS(mesh string) (*x509.CertPool, error) {
+	meshRes := core_mesh.NewMeshResource()
+	err := a.rm.Get(context.Background(), meshRes, core_store.GetByKey(mesh, model.NoMesh))
 	if err != nil {
-		return "", errors.Wrap(err, "unable to retrieve the signing key")
+		return nil, err
 	}
-
-	return string(key), nil
+	backend := meshRes.GetEnabledCertificateAuthorityBackend()
+	if backend == nil {
+		return nil, nil
+	}
+	caManager, ok := a.caManagers[backend.Type]
+	if !ok {
+		return nil, errors.Errorf("cannot find CA Manager for type %s", backend.Type)
+	}
+	rootCerts, err := caManager.GetRootCert(context.Background(), mesh, backend)
+	if err != nil {
+		return nil, err
+	}
+	certPool := x509.NewCertPool()
+	for _, certPEM := range rootCerts {
+		block, _ := pem.Decode(certPEM)
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, err
+		}
+		certPool.AddCert(cert)
+	}
+	return certPool, nil
 }
 
+const (
+	quitquitquit = "quitquitquit"
+)
+
 func (a *envoyAdminClient) adminAddress(dataplane *core_mesh.DataplaneResource) string {
 	ip := dataplane.GetIP()
 	// TODO: this will work perfectly fine with K8s, but will fail for Universal
 	// The real allocated admin port is part of the DP metadata, but it is attached to a particular CP,
 	// so we can not reliably use that. A better approach would be to include the admin port
 	// in the DataplaneInsights.
-	portUint := a.cfg.Runtime.Kubernetes.Injector.SidecarContainer.AdminPort
-
-	return net.JoinHostPort(ip, strconv.FormatUint(uint64(portUint), 10))
+	return net.JoinHostPort(ip, strconv.FormatUint(uint64(a.adminPort), 10))
 }
 
 func (a *envoyAdminClient) PostQuit(dataplane *core_mesh.DataplaneResource) error {
-	token, err := a.GenerateAPIToken(dataplane)
+	httpClient, err := a.buildHTTPClient(dataplane)
```

Comment:
Why it's preferable to build httpClient per request? Were there any problems with old approach when httpClient was built in constructor

Reply:
It's not preferable but we need to set CA of the Mesh-based on the dataplane.